python/taichi/lang/matrix_ops_utils.py

```diff
@@ -99,7 +99,23 @@ def check(*args, **kwargs):
 
 
 def same_shapes(*xs):
-    shapes = [x.get_shape() for x in xs]
+    shapes = []
+    for x in xs:
+        if isinstance(x, Matrix):
+            shapes.append(x.get_shape())
+        elif isinstance(x, list):
+            shape = [len(x)]
+            for i in x:
+                if isinstance(i, list):
+                    shape.append(len(i))
```

Comment:
Hmm will this result in `[3, 4, 4, 4]` for a 3 x 4 list?

Reply:
Ah thanks!